std/experimental/perpetual/perpetual.d

```diff
@@ -0,0 +1,352 @@
+
+		} else
+			static assert(0);
+	}
+
+	
+	private size_t fileSize(string path) {
+		version (Windows) {
+
+		} else version (Posix) {
+			stat_t st;
+			if(fstat(_fd, &st)) {
+				close(_fd);
+				errnoEnforce(false, _tag~"("~path~")");
+			}
+			return st.st_size;
+	
+		} else
+			static assert(0);
+	}
+
+	private void fileExpand(string path) {
+		version (Windows) {
+
+		} else version (Posix) {
+			lseek(_fd, T.sizeof-1, SEEK_SET);
+			char c=0;
+			core.sys.posix.unistd.write(_fd, &c, 1);
+		} else
+			static assert(0);
+	}
+
+}
+
+
+/**
+ * Convenience helper, equivalent to new Persistent!T
+ */
+auto perpetual(T)(string path)
+{
+	return new Perpetual!T(path);
+}
+
+
+/**
+ * Single argument helper,
+ *  uses copy ctor/assignment for initialization
+ */
+auto perpetual(T, U)(string path, U seed)
+{
+	auto p=new Perpetual!T(path);
+	if(p.master) {
+		static if(isArray!T) {
+			static if(isArray!U) {
+				auto l=min(T.length, seed.length);
+				p[0..l]=seed[0..l];
+			} else {
+				p[]=seed;
+			}
+		} else {
+			p=seed;
+		}
+	}
+	return p;
+}
+
+
+/**
+ * Multi-argument helper,
+ *  forwards arguments to ctor
+ */
+auto perpetual(T, Args...)(string path, Args args)
+{
+	auto p=new Perpetual!T(path);
+	if(p.master)
+		p=T(args);
+	return p;
+}
+
+
+///
+unittest
+{
+	import std.stdio;
+	import std.conv;
+	import std.string;
+	import std.getopt;
+	import std.perpetual;
+
+	struct A {
+		int x, y;
+		this(int x, int y) { this.x=x; this.y=y; }
+	}
+	class B {};
+	enum Color { black, red, green, blue, white };
+
+	// Perpetual!int                   : 0
+	// Perpetual!double                : nan
+	// Perpetual!(A)                   : A(1, 2)
+	// Perpetual!(int[5])              : [7, 5, 3, 1, 0]
+	// Perpetual!(immutable(short[]))  : [7, 0, 5, 0, 3, 0, 1, 0, 0, 0]
+	// Perpetual!(Color)               : white
+	// Perpetual!(char[])              : ________________________________
+	// Perpetual!string                : ________________________________
+	// Perpetual!(char[3][5])          : ["one", "...", "two", "...", "..."]
+	// Perpetual!(const(char[]))       : one...two......
+	// Perpetual!(char[3][])           : ["one", "...", "two", "...", "..."]
+	//
+	// Usage: test --int=3 --real=3.14159 --struct=11 --array=1,3,5,7 --color=green --string=ABCDE 
+	// Output:
+	// Perpetual!int                   : 3
+	// Perpetual!double                : 3.14159
+	// Perpetual!(A)                   : A(11, 2)
+	// Perpetual!(int[5])              : [1, 3, 5, 7, 0]
+	// Perpetual!(immutable(short[]))  : [1, 0, 3, 0, 5, 0, 7, 0, 0, 0]
+	// Perpetual!(Color)               : green
+	// Perpetual!(char[])              : ABCDE___________________________
+	// Perpetual!string                : ABCDE___________________________
+	// Perpetual!(char[3][5])          : ["one", "...", "two", "...", "..."]
+	// Perpetual!(const(char[]))       : one...two......
+	// Perpetual!(char[3][])           : ["one", "...", "two", "...", "..."]
+
+
+	void main(string[] arg)
```

Comment:
No main please. Use unittest. See also std.file unittests for `deleteme` and other ways to unittest with files